tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,197 @@
+        echo "Argument passed as configuration name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    configName="$3"
+    configValue="$(eval_tool "$repoRoot" "$toolName"; eval echo "\$$configName")"
+
+    if [ -z "$configValue" ]; then
+        echo "Unable to read the value corresponding to $configName from the .toolversions file."
+        exit 1
+    fi
+
+    echo "$configValue"
+}
+
+# Gets the name of the download package corresponding to the specified tool name.
+# Download package name is read from the .toolversions file.
+# Exit 1 if unable to read the name of the download package from the .toolversions file.
+get_download_package_name()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        usage
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    osName="$(get_os_name)"
+    get_tool_config_value "$repoRoot" "$toolName" "DownloadFile$osName"
+}
+
+# Gets the search path corresponding to the specified tool name.
+# Search path is read from the .toolversions file.
+# Exit 1 if unable to read the path from the .toolversions file.
+get_repository_tool_search_path()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        usage
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    osName="$(get_os_name)"
+    searchPath="$(get_tool_config_value "$repoRoot" "$toolName" "SearchPath${osName}Tools")"
+
+    echo "$repoRoot/$searchPath"
+}
+
+# Gets the error message to be displayed when the specified tool is not available for the build.
+# Error message is read from the .toolversions file.
+# Exit 1 if unable to read the error message from the .toolversions file.
+tool_not_found_message()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        usage
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
+
+    # Eval in a subshell to avoid conflict with existing variables.
+    (
+        eval_tool "$repoRoot" "$toolName"
+
+        if [ -z "$ToolNotFoundMessage" ]; then
+            echo "Unable to locate $toolName."
+            exit 1
+        fi
+
+        eval echo "$ToolNotFoundMessage"
+    )
+}
+
+# Write the given message(s) to probe log file.
+log_message()
+{
+    scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
```

Comment:
Ok. Do we write the log (probe-tool.log) to repo root or under Tools folder? I'm thinking repo root.

---
In reply to: [110423284](https://github.com/ravimeda/corefx/pull/9#discussion_r110423284) [](ancestors = 110423284)